src/axom/inlet/tests/inlet_errors.cpp

```diff
@@ -0,0 +1,350 @@
+// Copyright (c) 2017-2021, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#include "gtest/gtest.h"
+
+#include "axom/inlet/Inlet.hpp"
+#include "axom/inlet/tests/inlet_test_utils.hpp"
+
+using axom::inlet::Inlet;
+using axom::inlet::VerificationError;
+using axom::sidre::DataStore;
+
+template <typename InletReader>
+Inlet createBasicInlet(const std::string& luaString)
+{
+  std::unique_ptr<InletReader> reader(new InletReader());
+  reader->parseString(axom::inlet::detail::fromLuaTo<InletReader>(luaString));
+  const bool enableDocs = true;
+  return Inlet(std::move(reader), enableDocs);
+}
+
+template <typename InletReader>
+class inlet_errors : public ::testing::Test
+{ };
+
+TYPED_TEST_SUITE(inlet_errors, axom::inlet::detail::ReaderTypes);
+
+TYPED_TEST(inlet_errors, required_field)
+{
+  std::string testString = "";
+  Inlet inlet = createBasicInlet<TypeParam>(testString);
+
+  inlet.addInt("foo", "foo's description").required();
+
+  std::vector<VerificationError> errors;
+  EXPECT_FALSE(inlet.verify(&errors));
+  // Need something about "foo" and "Required"
+  EXPECT_TRUE(
+    std::any_of(errors.begin(), errors.end(), [](const VerificationError& err) {
+      return (static_cast<std::string>(err.path) == "foo") &&
```

Comment:
Suggestion: Some of the common checks might be good candidates for functions in the `VerificationError` struct.